edx_exams/apps/api/v1/tests/test_views.py

```diff
@@ -1752,3 +1753,101 @@ def test_exam_provider(self):
                 'proctoring_escalation_email': self.config.escalation_email,
             }
         )
+
+
+class AllowanceViewTests(ExamsAPITestCase):
+    """
+    Tests AllowanceView
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self.course_id = 'course-v1:edx+test+f19'
+        self.exam = ExamFactory(
+            course_id=self.course_id,
+        )
+
+    def request_api(self, method, user, course_id):
+        """
+        Helper function to make API request
+        """
+        assert method in ['get']
+        headers = self.build_jwt_headers(user)
+        url = reverse(
+            'api:v1:course-allowances',
+            kwargs={'course_id': course_id}
+        )
+
+        return getattr(self.client, method)(url, **headers)
+
+    def test_auth_required(self):
+        """
+        Test endpoint requires authentication and a staff user
+        """
+
+        # no auth
+        response = self.client.get(
+            reverse('api:v1:course-allowances', kwargs={'course_id': self.course_id}),
+        )
+        self.assertEqual(response.status_code, 401)
+
+        # no permissions
+        user = UserFactory.create(is_staff=False)
+        response = self.request_api('get', user, self.course_id)
+        self.assertEqual(response.status_code, 403)
+
+        # course staff has access
+        course_staff_user = UserFactory.create()
+        CourseStaffRole.objects.create(user=course_staff_user, course_id=self.course_id)
+        response = self.request_api('get', course_staff_user, self.course_id)
+        self.assertEqual(response.status_code, 200)
+
+    def test_get_allowances(self):
```

Comment:
Should we have a DB query count check? I want to ensure this API only call DB once or twice, for even 1000 returned records.

Reply:
IMO we only need to check query counts if is there's something more interesting happening in the business logic or model that might hide DB issues. In this case it's a pretty straightforward use of `.filter()`